sdks/python/src/opik/evaluation/metrics/heuristics/bleu.py

```diff
@@ -0,0 +1,279 @@
+from typing import Any, List, Optional, Tuple, Union
+
+from opik.evaluation.metrics.exceptions import MetricComputationError
+from opik.evaluation.metrics import base_metric, score_result
+
+try:
+    from nltk.translate import bleu_score as nltk_bleu_score
+except ImportError:
+    nltk_bleu_score = None
+
+
+class BaseBLEU(base_metric.BaseMetric):
+    """
+    Base class containing shared BLEU logic, such as handling n-grams, smoothing,
+    and weights initialization. This class is not intended to be used directly.
+
+    References:
+      - NLTK BLEU smoothing:
+        https://www.nltk.org/api/nltk.translate.bleu_score.html#nltk.translate.bleu_score.SmoothingFunction
+
+    Args:
+        name: The name of the metric (e.g. "sentence_bleu_metric" or "corpus_bleu_metric").
+        track: Whether to track the metric (depends on your system).
+        n_grams: Up to which n-gram order to use (1 through n_grams).
+        smoothing_method: One of NLTK's SmoothingFunction methods (e.g. "method0", "method1", etc.).
+        weights: Optional custom weights for n-gram orders. Must sum to 1.0. If None,
+                 defaults to uniform distribution across `n_grams`.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        track: bool,
+        n_grams: int,
+        smoothing_method: str,
+        weights: Optional[List[float]],
+    ):
+        super().__init__(name=name, track=track)
+
+        if nltk_bleu_score is None:
+            raise ImportError(
+                "`nltk` library is required for BLEU score calculation. "
+                "Install via `pip install nltk`."
+            )
+
+        self.n_grams = n_grams
+        self.smoothing_method = smoothing_method
+
+        if weights is None:
+            self.weights = [1.0 / n_grams] * n_grams
+        else:
+            if len(weights) != n_grams:
+                raise ValueError(
+                    f"Length of weights ({len(weights)}) != n_grams ({n_grams})."
+                )
+            if abs(sum(weights) - 1.0) > 1e-6:
+                raise ValueError("Weights must sum to 1.0")
+            self.weights = weights
+
+        self._smoother = nltk_bleu_score.SmoothingFunction()
+
+    def _get_smoothing_func(self) -> nltk_bleu_score.SmoothingFunction:
```

Comment:
Looks good, thank you!

Only one thing after I ran the tests, we need to rename
`nltk_bleu_score.SmoothingFunction -> "nltk_bleu_score.SmoothingFunction"`
That's why the e2e tests are failing, they don't have `nltk` package dependency and can't import this module.

You can ignore the rest of the errors in CI except for the unit tests and e2e tests, other workflows are red just just because they require different API keys.